runtime/near-evm-runner/src/builtins.rs

```diff
@@ -598,4 +643,426 @@ impl Impl for Blake2FImpl {
         output.write(0, &output_buf[..]);
         Ok(())
     }
+
+    fn gas(&self, input: &[u8], precompile_costs: &EvmPrecompileCostConfig) -> U256 {
+        Pricing::Blake2F(precompile_costs.blake2f_cost).cost(input)
+    }
+}
+
+// Below is part of Open Ethereum:
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
+
+// Open Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Open Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Open Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+/// A gas pricing scheme for built-in contracts.
+trait Pricer: Send + Sync {
+    /// The gas cost of running this built-in for the given input data at block number `at`
+    fn cost(&self, input: &[u8]) -> U256;
+}
```

Comment:
Can you move `Pricer` trait and implementations to a separate mod?

Reply:
done